test/integration/kubernetes_deploy_test.rb

```diff
@@ -384,8 +422,27 @@ def test_output_when_switching_labels_incorrectly
       web["spec"]["template"]["metadata"]["labels"] = { "name" => "foobar" }
     end
     assert_equal false, success
-    assert_logs_match("one of your templates is invalid")
-    assert_logs_match(/The Deployment "web" is invalid.*`selector` does not match template `labels`/)
+
+    assert_logs_match_all([
+      "one of your templates is invalid",
+      /The Deployment "web" is invalid.*`selector` does not match template `labels`/
+    ])
+  end
+
+  def test_can_deploy_deployment_with_zero_replicas
+    success = deploy_fixtures("hello-cloud", subset: ["configmap-data.yml", "web.yml.erb"]) do |fixtures|
+      web = fixtures["web.yml.erb"]["Deployment"].first
+      web["spec"]["replicas"] = 0
```

Comment:
Yes, that should work fine. [The implementation](https://github.com/Shopify/kubernetes-deploy/pull/107/files#diff-bfab9900d04f0a61190aa04f896aed1fR50) uses selectors to grab a live copy of the related deployment's template, so it should work even if the deployment wasn't managed by `kubernetes-deploy`.